wrappa/api_auth_wrappa.go

```diff
@@ -140,12 +142,17 @@ func (wrappa *APIAuthWrappa) Wrap(handlers rata.Handlers) rata.Handlers {
 			panic("you missed a spot")
 		}
 
-		if name == atc.GetAuthToken {
+		if name == atc.GetAuthToken || name == atc.GetAccessToken {
 			newHandler = auth.WrapHandler(newHandler, wrappa.getTokenValidator, wrappa.userContextReader)
 		} else {
 			newHandler = auth.WrapHandler(newHandler, wrappa.authValidator, wrappa.userContextReader)
 		}
-		wrapped[name] = auth.CSRFValidationHandler(newHandler, rejector, wrappa.userContextReader)
+		if name != atc.CheckResourceWebHook {
```

Comment:
Without this special case the webhook caller would have to include the `X-Csrf-Token` header on each request and shouldn't be available for the caller to access otherwise the CSRF protection becomes useless. So thus we added the special case. We wanted to do it more generically but that would have been a bigger change.